charts/kong/templates/_helpers.tpl

```diff
@@ -116,12 +117,74 @@ spec:
           {{- if (not (eq .ingressVersion "extensions/v1beta1")) }}
           pathType: {{ $pathType }}
           {{- end }}
+  {{- else if $hostname }}
+  - host: {{ $hostname | quote }}
+    http:
+      paths:
+        - backend:
+          {{- if (not (eq .ingressVersion "networking.k8s.io/v1")) }}
+            serviceName: {{ .fullName }}-{{ .serviceName }}
+            servicePort: {{ $servicePort }}
+          {{- else }}
+            service:
+              name: {{ .fullName }}-{{ .serviceName }}
+              port:
+                number: {{ $servicePort }}
+            {{- end }}
+          path: {{ $path }}
+          {{- if (not (eq .ingressVersion "extensions/v1beta1")) }}
```

Comment:
You might want to drop the v1beta1 branch here since support for v1beta1 ingresses has been dropped in KIC 2.10: https://github.com/Kong/kubernetes-ingress-controller/blob/v2.10.0/CHANGELOG.md#deprecated

Reply:
Sure, I can remove the ingress version related things.

Reply:
Done.